### DISCUSSION TITLE
如何通过ingress让外部可以访问milvus

### DISCUSSION THREAD

**Discussion by ghost** [milvus-io/milvus#24299] [Q&A and General discussion]
milvus是用 k8s 搭建的集群版本，通过 svc 访问全部正常，现在想让非 k8s 节点也访问，于是手动配置 ingress，具体配置如下:

`apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    nginx.ingress.kubernetes.io/cors-allow-headers: DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization,openID,audiotoken
    nginx.ingress.kubernetes.io/enable-cors: 'true'
    nginx.ingress.kubernetes.io/ssl-redirect: 'false'
  name: milvusgrpc-ingress
  namespace: milvus
spec:
  rules:
  - host: milvusgrpc.test.com.cn
    http:
      paths:
      - backend:
          service:
            name: test-milvus
            port:
              number: 19530
        path: /
        pathType: Prefix`

现在 telnet milvusgrpc.test.com.cn 80  没问题，但用 python sdk 访问 这个地址，直接超时，不知道如何排错，希望大佬指点下，多谢

**Comment by LoveEachDay** [comment]
@580825 what kind of ingress controller do you use?  And you can check the ingress controller log to verify wheher the ingress config is valid.

**Comment by ghost** [comment]
@LoveEachDay 

ingress controller:    ingress-nginx-controller
k8s version:              v1.26.2

try curl:

`
curl milvusgrpc.test.com
curl: (1) Received HTTP/0.9 when not allowed

nginx controller log:

2023/05/23 01:29:09 [error] 2168#2168: *5029086 upstream sent no valid HTTP/1.0 header while reading response header from upstream, client: 10.12.6.66, server: milvusgrpc.test.com.cn, request: "GET / HTTP/1.1", upstream: "http://10.44.14.14:19530/", host: "milvusgrpc.test.com.cn"
2023/05/23 01:29:09 [error] 2168#2168: *5029086 recv() failed (104: Connection reset by peer) while reading upstream, client: 10.12.6.66, server: milvusgrpc.test.com.cn, request: "GET / HTTP/1.1", upstream: "http://10.44.14.14:19530/", host: "milvusgrpc.test.com.cn"
10.12.6.66 - - [23/May/2023:01:29:09 +0000] milvusgrpc.test.com.cn "GET / HTTP/1.1" 009 15 "-" "curl/7.76.1" 92 0.002 [milvus-test-test-milvus-19530] [] 10.44.14.14:19530 15 0.001 200 d64db7d11de6bab780
`

try python sdk:
`
from pymilvus import (
    connections,
    utility,
    FieldSchema,
    CollectionSchema,
    DataType,
    Collection,
)

connections.connect("default", host="milvusgrpc.test.com", port="80", secure=False)


Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pymilvus/client/grpc_handler.py", line 116, in _wait_for_channel_ready
    grpc.channel_ready_future(self._channel).result(timeout=timeout)
  File "/usr/local/lib64/python3.9/site-packages/grpc/_utilities.py", line 151, in result
    self._block(timeout)
  File "/usr/local/lib64/python3.9/site-packages/grpc/_utilities.py", line 97, in _block
    raise grpc.FutureTimeoutError()
grpc.FutureTimeoutError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "//abc.py", line 10, in <module>
    connections.connect("default", host="milvusgrpc.test.com", port="80", secure=False)
  File "/usr/local/lib/python3.9/site-packages/pymilvus/orm/connections.py", line 312, in connect
    connect_milvus(**kwargs, user=user, password=password)
  File "/usr/local/lib/python3.9/site-packages/pymilvus/orm/connections.py", line 271, in connect_milvus
    gh._wait_for_channel_ready(timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/pymilvus/client/grpc_handler.py", line 119, in _wait_for_channel_ready
    raise MilvusException(Status.CONNECT_FAILED,
pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on milvusgrpc.test.com:80. Timeout)>

`

nginx controller does not have any log

**Comment by Mararesliu** [comment]
the same problem, i use dns domain can deal with it. But any tcp grpc loadbalancer can not pass . Any Update ?
